src/main/java/Controller/musicEngineController.java

```diff
@@ -59,21 +58,23 @@ public void playSpace(){
 
     /**
      * function calling the use case for adding a song to the space
-     * @param songInputData input data for adding a song to the space
+     * @param songToAddID id of song to add to space
      */
-    public void spaceAddSong(songInputData songInputData){
-        int songToAddID = songInputData.getSong().getID();
+    public void spaceAddSong(int songToAddID){
         for (songInputData currSongInputData : this.spaceSongList){
             int currSongID = currSongInputData.getSong().getID();
             if (currSongID == songToAddID){
+                this.spacePresenter.notAddedToSpace(currSongInputData.getName());
```

Comment:
This seems like it could be use case code which might be better in a use case interactor, but we don't have an addToSpaceInteractor. I don't think this is an issue because of the way we are dealing with spaces. If it was a singleton, it might've made sense to create an interactor for it but since we are using space as a List of songs in the controller (as per the profs reccomendation), it doesn't make sense to have to keep passing around the List of songs. Although it's not typical for the controller to talk to to presenter, I think it's fine in this case since they're both in the same layer so it shouldn't be a violation of clean architecture.

Reply:
To expand a bit on prof's recommendation, we got rid of the addToSpace use case as a whole because adding song to space does not require reaching to the back end. (because there is no specific rule we're just adding to the list). And yes I agree that this does not violate clean architecture because the presenter and the controller are in the same layer and thus allowed to talk to each other.